### PR TITLE
Veterans - Eligibility Verifier + Eligibility Type

### LINKED ISSUE #1426: Define EligibilityType for veterans
Starting with MST for launch. Link to Payment Enrollment group.

### LINKED ISSUE #1427: Define EligibilityVerifier for veterans
This model holds references to most of the content that is displayed for the Veterans flow. Focus on English copy only for the basic/existing UI (i.e. no modals etc.)

### PR DESCRIPTION
**Pull request by machikoyasuda**
closes #1427 
closes #1426

## What this PR does
- [x] Create `EligibilityVerifier` for MST Veterans
- [x] Create `EligibilityType`  for MST Veterans
- [x] Create Terraform values stubs for:  `"MST_VETERAN_GROUP_ID"`, `"MST_VETERAN_VERIFIER_NAME"`
- [x] Spec: Update `benefit-select.cy.js` spec from looking for 2 radio inputs, to 3 inputs. Update Cypress helper method so it correctly chooses Courtesy Card radio button.
- [x] Copy: Eligibility Index page and add VA.gov url as well (but not the new external link icon design).
- [x] Copy: Enrollment Success page - Same as Login.gov 
- [x] Copy: Eligibility Start page: 2 paragraphs (except for the required_items bullet list - that's not part of `EligibilityVerifier`) 
- [x] UI: Eligibility Index page - Make sure US Veteran option is the 2nd choice.
- [ ] #1426 

## What configuration has been done
- [x] Update KeyVault development for EligibilityVerifier: Create and set these key/values
- [x] Update KeyVault test for EligibilityVerifier: Create and set these key/values
- [x] Create Veteran group in LittlePay
- [x] Update KeyVault development for EligibilityType
- [x] Update KeyVault test for EligibilityVerifier

## Screenshots

image
image

### EligibilityVerifier model
- Adds `start_item_secondary_details`, an optional `TextField()`, to `EligibilityVerifier` model

### MediaItem viewmodel
- Adds `secondary_details`, an optional string, to `MediaItem` viewmodel class

The `secondary_details` is this details paragraph that comes after the bullets, on the Eligibility Start page.

image

## Notes

On Production, after going through the Login.gov flow and verifying eligibility, the user sees the `enrollment` screen with this copy, consisting of 2 media items:
image

In the new design, however, there is only 1 media item for all `enrollment` screens, and the remaining media item for Little Pay is also shortened from 2 paragraphs to 1:
image

### CODE REVIEW

**Comment by machikoyasuda** [comment]
@thekaveman @angela-tran I'm blocked here: https://github.com/cal-itp/benefits/issues/1437 Do you see any syntax error or something I'm doing wrong?

File: tests/cypress/plugins/helpers.js
Comment (machikoyasuda) [inline tests/cypress/plugins/helpers.js:28]: This TODO apt is very appropriate 😅  Ideally the code would say something like: "Look for the Courtesy Card form input text", "Select the input closest to the Courtesy Card text"

Comment (machikoyasuda) [reply tests/cypress/plugins/helpers.js:28]: Should create a ticket for the cooldown time to refactor this method.

File: terraform/app_service.tf
Comment (machikoyasuda) [inline terraform/app_service.tf:100]: @thekaveman @angela-tran This is all I did in this `app_service.tf` file. Am I supposed to configure something manually in KeyVault after this PR is merged to define some test values.. or something?

File: benefits/core/migrations/0002_data.py
Comment (machikoyasuda) [inline benefits/core/migrations/0002_data.py:20]: @angela-tran @thekaveman Is this all we need for #1426? Does this PR also close #1426 or is there something more that it entails? Wasn't sure.

Comment (angela-tran) [reply benefits/core/migrations/0002_data.py:20]: I think #1426 also includes creating the group in Littlepay and adding the group ID to Key Vault in Azure for dev and test.

File: terraform/app_service.tf
Comment (angela-tran) [reply terraform/app_service.tf:100]: Yeah, you can configure the value in Key Vault even before this PR is merged. You'll create a new secret called `mst-veteran-verifier-name` with a value similar to the names for other EligibilityVerifiers.

File: benefits/core/migrations/0002_data.py
Comment (angela-tran) [inline benefits/core/migrations/0002_data.py:176]: I wonder if we should rename this PO entry since it is now being used for the Login.gov `EligibilityVerifier` and the veteran `EligibilityVerifier`? 

Also, I'm looking at how we have a [different PO entry being used for Courtesy Cards](https://github.com/cal-itp/benefits/pull/1436/files#diff-f4aa1ce24b8c65ccf8b80dacab3cb3aff3e95d9bf430e8b22bc66eb233f40b23R215) ([`eligibility.pages.unverified.mst_cc.title`](https://github.com/cal-itp/benefits/blob/dev/benefits/locale/en/LC_MESSAGES/django.po#L121)), but... does the "Developer Benefits Full App" page in Figma imply now that all `unverified_title` and `unverified_blurb`s should be the same?

image

image


cc @srhhnry @thekaveman

Comment (angela-tran) [reply benefits/core/migrations/0002_data.py:176]: Side thought -- this just gave me an idea for something to try during dev-design handoff (I worry I'll forget it if I wait until retro): 

As a part of design-dev handoff, could we go ahead and make GitHub tickets at that point to capture the changes that need to be done?

Comment (machikoyasuda) [reply benefits/core/migrations/0002_data.py:176]: Yeah and now the enrollment page copy is all the same - so there's some more copy string consolidation that can be done. But not sure what's in-scope vs. out-of-scope in this ticket and in this Sprint overall.

File: benefits/eligibility/views.py
Comment (angela-tran) [inline benefits/eligibility/views.py:109]: Noticed a typo
```suggestion
        secondary_details=_(verifier.start_item.secondary_details),
```

Comment (angela-tran) [reply benefits/eligibility/views.py:109]: Oops, never mind! Got mixed up between the field and the PO msgid

**Review by angela-tran — COMMENTED** [review]
These changes look good to me @machikoyasuda! I'm inclined to approve, but would like to get others' feedback about the PO entry naming / consolidation. Maybe at the very least we can remove `login_gov` from that PO entry msgid and leave the Courtesy Card verifier alone for now.

File: benefits/locale/en/LC_MESSAGES/django.po
Comment (machikoyasuda) [inline benefits/locale/en/LC_MESSAGES/django.po:89]: should be `You selected a Veteran transit benefit.`

File: benefits/core/migrations/0002_data.py
Comment (thekaveman) [reply benefits/core/migrations/0002_data.py:20]: This change seems out of scope for this PR? Why not leave it all for the PR that closes #1426?

File: terraform/app_service.tf
Comment (machikoyasuda) [reply terraform/app_service.tf:100]: Added `mst-veteran-verifier-name` to Key Vault (Dev).

**Comment by machikoyasuda** [comment]
@thekaveman @angela-tran Ready for re-review, hopefully final review. I added a few changes since the last review from Angela:

1. The copy on Eligibility Start was off by one word.
1. Removed `login_gov` from `msgid`s for Unverified page, b/c all the Unverified pages have the same copy now.
1. Added VA.gov link (which required adding `format_html()` code)

File: benefits/core/migrations/0002_data.py
Comment (thekaveman) [reply benefits/core/migrations/0002_data.py:20]: @machikoyasuda it looks like you did fold #1426 into this one (creating the LP group, configuring it in KeyVault, linking to the EligibilityType)?

File: tests/cypress/plugins/helpers.js
Comment (thekaveman) [reply tests/cypress/plugins/helpers.js:28]: #1443

**Review by thekaveman — APPROVED** [review]
Please update the description to also note that this PR closes #1426 

Otherwise this looks good and can be approved once the Test KeyVault secrets are created!

File: benefits/core/migrations/0002_data.py
Comment (machikoyasuda) [inline benefits/core/migrations/0002_data.py:166]: @thekaveman Should I just put some other variable here then? Like just stub it with `mst_senior_type`?

File: benefits/core/migrations/0002_data.py
Comment (machikoyasuda) [reply benefits/core/migrations/0002_data.py:20]: I just wasn't really sure what to put here otherwise: https://github.com/cal-itp/benefits/pull/1436/files#r1238987028

File: benefits/core/migrations/0002_data.py
Comment (thekaveman) [reply benefits/core/migrations/0002_data.py:166]: No, I think it's fine that this PR closes both (and I see the reasoning) -- just want to make sure it fully closes both (LP config, KeyVaults, etc.)

**Comment by machikoyasuda** [comment]
@angela-tran No not yet. 

This is my current status:

image

File: benefits/core/migrations/0002_data.py
Comment (thekaveman) [reply benefits/core/migrations/0002_data.py:20]: https://github.com/cal-itp/benefits/pull/1436#discussion_r1238989101

**Comment by machikoyasuda** [comment]
- Created MST Veteran group on QA (Documented here https://github.com/cal-itp/benefits-secrets/pull/36)
- Updated KeyVault Dev with the new group id
- Updated KeyVault Test with the new group id and the verifier name

@angela-tran @thekaveman 

Ready for final...final 😅 ...review

**Review by angela-tran — APPROVED** [review]
Looks good to me @machikoyasuda 👍

**Comment by thekaveman** [comment]
Ah @machikoyasuda I was still reviewing this!

File: benefits/locale/en/LC_MESSAGES/django.po
Comment (thekaveman) [inline benefits/locale/en/LC_MESSAGES/django.po:8]: Why did we lose our Issues link here?

Comment (thekaveman) [reply benefits/locale/en/LC_MESSAGES/django.po:8]: It looks like the helper `bin/makemessages.sh` was not run, it is supposed to put this line back.

**Comment by machikoyasuda** [comment]
@thekaveman Oops sorry!!! 😓 

I just noticed I get this error after running `bin/makemessages.sh` now:

```
calitp@cdbe8faf8c05:~/app$ bin/makemessages.sh
processing locale es
processing locale en
sed: couldn't open temporary file benefits/locale/en/LC_MESSAGES/sed3E0pRF: Permission denied
```

which fails when running `sed -i 's/Report-Msgid-Bugs-To\:/Report-Msgid-Bugs-To\: https\:\/\/github.com\/cal-itp\/benefits\/issues/g' benefits/locale/*/*/*.po` - so that's why that line failed to get changed.

**Comment by machikoyasuda** [comment]
Reported here: https://github.com/cal-itp/benefits/issues/1455

**Pull request merged**
